tests/Behat/FeatureContext.php

```diff
@@ -298,4 +298,30 @@ public function assertVisibility($element): void {
     $this->assertVisuallyVisible($node);
   }
 
+  /**
+   * Creates stakeholder/s for organisation entity.
+   *
+   * Usage example:
+   *
+   * Given the following stakeholders:
+   *   | name   |
+   *   | name 2 |
+   *   |   ...  |
+   *
+   * @Given the following stakeholder(s):
+   */
+  public function givenStakeholder(TableNode $table) {
+    $stakeholders = $table->getColumnsHash();
+    $storage = \Drupal::service('entity_type.manager')->getStorage('oe_organisation');
+
+    foreach ($stakeholders as $stakeholder) {
+      $entity = $storage->create([
+        'bundle' => 'oe_stakeholder',
+        'name' => $stakeholder['name'],
+        'status' => 1,
+      ]);
+      $entity->save();
+    }
+  }
+
```

Comment:
We need to provide a corporate reusable context, check what we did here:
```
./tests/Behat/Content/Venue/DefaultVenueContext.php
./tests/Behat/Content/Contact/PressContactContext.php
./tests/Behat/Content/Contact/GeneralContactContext.php
```

Reply:
refactored

Reply:
Sorry @drishu, what I meant was to add a specific context for the stakeholder organisation bundle, like those 3 above, so something named like:

```
./tests/Behat/Content/Organisation/StakeholderOrganisationContext.php
```
Then in the `ProjectContentContext` you created we reference those entities.

Check what we did for event and contact entities.

Reply:
Is it possible to add a commun function to manage same behavior and not copy paste the same code :

```
foreach ($scope->getFields() as $key => $value) {

// Handle entity references.
switch ($key) {
  case 'Published':
    $scope->addFields([
    $mapping[$key] => (int) ($value === 'Yes'),
  ])->removeField($key);
  break;

  default:
    if (isset($mapping[$key])) {
      $scope->renameField($key, $mapping[$key]);
    }
  }
}
```
?


For that, I suggest to add the following function in `tests/Behat/Content/CorporateContentContext.php` :

```
/**
 * Run before fields are parsed by Drupal Behat extension.
 *
 * @param \Drupal\Tests\oe_content\Behat\Hook\Scope\BeforeParseEntityFieldsScope $scope
 *   Behat hook scope.
 *
 * @BeforeParseEntityFields(oe_organisation,oe_stakeholder)
 */
public function mapFieldsLabelName(BeforeParseEntityFieldsScope $scope): void {
  // Maps human readable field names to their Behat parsable machine names.
  $bundle_fields = \Drupal::getContainer()->get('entity_field.manager')->getFieldDefinitions($scope->getEntityType(), $scope->getBundle());
  // var_dump(array_keys($bundle_fields));
  $mapping = [];
  foreach($bundle_fields as $field_name => $field_config){
    if(is_string($label = $field_config->getLabel())){
    $mapping[$label] = $field_name;
    }
  }
  $mapping['Published'] = 'status';
  
  foreach ($scope->getFields() as $key => $value) {
    switch ($key) {
      case 'Published':
        $scope->addFields([
          $mapping[$key] => (int) ($value === 'Yes'),
        ])->removeField($key);
        break;

      default:
        if (isset($mapping[$key])) {
          $scope->renameField($key, $mapping[$key]);
        }
    }
  }
}
```